packages/styleguide/src/components/Preview/Preview.js

```diff
@@ -26,310 +26,267 @@ import Button from './../Button';
 
 const CLASS_ROOT = '';
 
-function getBackgroundsAsArray(previewBackgrounds, excludedColors = []) {
-  return Object.keys(previewBackgrounds)
-    .filter(colorName => !excludedColors.includes(colorName))
-    .map(key => ({
-      value: previewBackgrounds[key],
-      label: key,
-    }));
-}
-
-class Preview extends Component {
-  static displayName = 'Preview';
-
-  static propTypes = {
-    /** Pass custom code (JSX) which will be shown instead of visually previewed code. */
-    code: oneOfType([node, func]),
-    /** Pass [react-element-to-jsx-string](https://github.com/algolia/react-element-to-jsx-string) options to code preview. */
-    codeJSXOptions: object,
-    /** Default background color as color name. Must be one from passed `bgThemeColors`. To disable background color chooser, pass `false`. */
-    bgTheme: oneOfType([string, bool]),
-    /** Available background colors. Colors are inherithed from `theme.previewBackgrounds`. */
-    bgThemeColors: object,
-    /** Exclude colors from available background colors. */
-    bgThemeExcludedColors: arrayOf(string),
-    /** Disables code preview. */
-    hasCodePreview: bool,
-    /** Pass HTML as string to preview HTML code. */
-    html: string,
-    /** (unstable) Display preview in inframe. */
-    isIframe: bool,
-    /** (unstable) Iframe custom <head /> */
-    iframeHead: node,
-    /** (unstable) Iframe custom JavaScripts. */
-    iframeScripts: string,
-    /** Preview with interactivity */
-    isInteractive: bool,
-    /** Props for interactive component */
-    interactiveProps: object,
-    /** Enable fullscreen toggle */
-    enableFullscreen: bool,
-  };
-
-  static defaultProps = {
-    bgTheme: 'white',
-    bgThemeColors: {
-      white: '#fff',
-    },
-    bgThemeExcludedColors: [],
-    hasCodePreview: true,
-    // to discuss if it should be turned on or off
-    // for debugging is better on
-    isInteractive: true,
-    interactiveProps: {},
-    enableFullscreen: true,
-  };
-
-  componentWillReceiveProps(props) {
-    if (
-      this.state.previewBackground &&
-      props.bgTheme !== this.state.previewBackground.label
-    ) {
-      this.setState({
-        previewBackground: {
-          label: props.bgTheme,
-          value: props.bgThemeColors[props.bgTheme],
-        },
-      });
-    }
-  }
-
-  constructor(props) {
-    super(props);
-
-    this.handleToggleCode = this.handleToggleCode.bind(this);
-    this.handleToggleInteract = this.handleToggleInteract.bind(this);
-    this.handlePreviewBackground = this.handlePreviewBackground.bind(this);
-    this.handleToggleFullscreen = this.handleToggleFullscreen.bind(this);
-  }
-
-  state = {
-    isCodeShown: false,
-    isInteractive: false,
-    previewBackground: this.props.bgThemeColors
+const getBackgroundsAsArray = (previewBackgrounds, excludedColors = []) =>
```

Comment:
Preco sa telo tejto funkcie prepisovalo z `.filter.map` na `.reduce`?

Reply:
prislo mi rozumnejsie prechadzat pole iba raz a rovno to tam checkovat

Reply:
btw neviem ci to bolo spravne ale aspon to pomohlo v tom ze beni mi povedal ze konecne pochopil co ten reduce v skotocnosti robi a ako ho pouzivat.

Reply:
Akoze na pochopenie reducue super :) Ale teda performance zlepsenie v tejto situacii je naozaj minimalny a zanedbatelny a myslim ze `.filter.map` je viac citatelne, co je sktutocny benefit v tejto situacii.

Reply:
je pravda že ak niekto nepozná reduce je to menej čitateľné. ale mne príde reduce ako veľmi dobrý nástroj na zredukovanie poľa podľa nejakých podmienok. asi som ho už využíval veľa tak som si zvykol. určite je možné že pre niekoho to bude čitateľnejšie :) bol to len môj úlet asi.